### ISSUE TITLE
Old allowed / blocked connection indicator

### ISSUE THREAD

**Issue by northys** [safing/portmaster-ui#195]
**What would you like to add or change?**:

I would like to vote for old good allowed / blocked connection indicator instead this weird looking progress bar.

IIRC the old one showed 2 numbers one with red dot for blocked connections and the other one with green dot for allowed connections. Since you changed it it's much harder to notice that portmaster is blocking something. For example for firefox where there are thousands of connections and most of them are allowed only a 1 px gray line is showed. This is almost unnoticeable.

**At least change the grey color to red, please!** So red that it increases my blood pressure immediately. That would allow me to notice blocked connection  easier.

**Why do you and others need this?**:

I find new design to be less usable than the old one. I already complained in e-mail back when you released it and @davegson had to explain me the new design because I was completely lost about what is the progress bar trying to tell me.

**Comment by davegson** [comment]
> IIRC the old one showed 2 numbers one with red dot for blocked connections and the other one with green dot for allowed connections.

There was only one number, counting the total connections. It was only red/green if 100% of all the connections was blocked/allowed.

That was what made it super useless for 99% of the time. Have a look:

![old_indicator](https://user-images.githubusercontent.com/3080765/148234634-6429b708-9382-432d-b318-c540d6093c13.png)

Compared to the previous version I really like the new bar indicator. It immediately gives an overview over all apps with a total connections count and a rough percentage of the allowed connections.

![bar_indicator](https://user-images.githubusercontent.com/3080765/148236359-76b36dad-8bda-4c6a-8ce1-8448fe296db3.png)

I understand a small gray line is not saying much in your case, with FF having hundreds of connections. But when the grey line is there Portmaster _is_ blocking something. You should notice the difference between full green bars and almost full green bars. Maybe we should make the min grey part bigger?

When many connections are involved, I feel it is required to move deeper for the details in any way. If you click on Firefox, you better see what is happening, with the amount of blocked connections shown and the possibility to filter by blocked connections.

> At least change the grey color to red, please! So red that it increases my blood pressure immediately. That would allow me to notice blocked connection easier.

This is tricky, since (1) the UI is colorful already as it is and (2) where should we put in failed connections? Mixing in a third color would make it overkill for sure. Also I think we tried it with red, but it is not pretty. It becomes very aggressive and not a pleasure to look at. IMO gray does the job just as well.

> I find new design to be less usable than the old one.

Do you still feel he same after given the old screenshot as context?

**Comment by northys** [comment]
@davegson ok I recalled it badly. Shame on me :D Maybe because I mostly fixed stuff that was 100% blocked like ssh, dlna and other apps that communicate only in home network. I don't whitelist stuff for firefox much so maybe that's the reason why I recall the red color.

I still don't understand why do you think red is colorful. have you ever seen anything that is green for success and gray for failure? I haven't. I would notice the red even if it would be 1% od the progress bar. Grey vanishes in the UI since most of it is greyscale.

Does it even work for colorblind person? I don't know how they see green grey and red. I recall that colorblind mode in WoW and LoL games swapped green and red for blue / red so they can definitely recognize red. I'm not color blind and I don't notice something is blocked most of the time.

Maybe my problem aren't the UI colors or the UI itself at all but that most of the time when something doesn't work at all or partly it takes me too much time before I realize something is blocked by portmaster.

I used to blame binance.com because their trading UI werent reloading in realtime for me in any browser and weeks later I realised that portmaster blocks only their websocket service. That's maybe also my fault because crypto is blocked by default and I've allowed binance.com only and not ws.binance.com.

TLDR it's hard for me to realise that portmaster blocked something.

**Comment by davegson** [comment]
Heya, this got lost a bit... Anyway, thanks for your humor :D

Given that this suggestion is about bringing back the old connection indicator, which are not making a comeback, I will close this issue.

-----

In terms of replacing grey with red, I would deny that as well for now, as it is:

- misleading, a user might interpret that the Portmaster blocks something and be angry about it when in reality a connection just failed.
- a step back for colorblind people. You correctly mention the alternative, but we are not at a stage where we tackle accessibility too much and would rather re-visit everything properly at a later point and until then not make temp-fixes or make things worse than they already are
- and again, it adds even more colors, which is debatable I know 😄

-----

Anyway, thanks for writing out the core issue behind all of this:

> TLDR it's hard for me to realize that portmaster blocked something and for some reason I recall it used to be better but maybe it wasn't. It's already a few months since you changed it.

And this is definitely something we want to improve! I think it's best to continue discussing this aspect after the work in https://github.com/safing/portmaster-ui/pull/204 has been released

Have a good one!
